extras/travis_build.sh

```diff
@@ -46,21 +46,26 @@ else
 fi
 
 if [ "$xPYTHON" = "1" ] ; then
-    if [ "$TRAVIS_PYTHON_VERSION" = "3.6.10" ]; then
-	PYPREFIX="/opt/python/3.6.10"
-	CONFIGURE_PYTHON="\
-            -DPYTHON_LIBRARY=$PYPREFIX/lib/libpython3.6m.so
-            -DPYTHON_INCLUDE_DIR=$PYPREFIX/include/python3.6m/"
-    fi
-    if [[ $OSTYPE = darwin* ]]; then
-	PYPREFIX="/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8"
-	CONFIGURE_PYTHON="\
-            -DPYTHON_LIBRARY=$PYPREFIX/lib/libpython3.8.dylib
-            -DPYTHON_INCLUDE_DIR=$PYPREFIX/include/python3.8"
-    fi
-
-    mkdir -p $HOME/.matplotlib
-    echo "backend : svg" > $HOME/.matplotlib/matplotlibrc
+   #PYTHON_LIB_DIR=`python3 -c "import sysconfig; print(sysconfig.get_path('platlib'))"`
+   PYTHON_LIB_DIR=`python3 -c "from distutils.sysconfig import get_config_var; print("%s/%s" % (get_config_var("LIBDIR"), get_config_var("INSTSONAME"))"'
+   echo "-> Detected PYTHON_LIB_DIR=$PYTHON_LIB_DIR"
+   ls "$PYTHON_LIB_DIR"
+   PYTHON_INCLUDE_DIR=`python3 -c "import sysconfig; print(sysconfig.get_path('include'))"`
+   echo "-> Detected PYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR"
+   ls "$PYTHON_INCLUDE_DIR"
+   if [ "$TRAVIS_PYTHON_VERSION" = "3.8.2" ]; then
+       #PYTHON_LIB_DIR=`find /usr/lib/x86_64-linux-gnu -name "libpython3*.so"`
+      PYTHON_LIBRARY=`find $PYTHON_LIB_DIR -name "libpython3*.so"` 
+   fi
+   if [[ $OSTYPE = darwin* ]]; then
+      #PYPREFIX="/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8"
+      PYTHON_LIBRARY=`find $PYTHON_LIB_DIR -name "libpython3.8.dylib"`
+   fi    
+   CONFIGURE_PYTHON="\
+       -DPYTHON_LIBRARY=$PYTHON_LIBRARY
+       -DPYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR"
+   mkdir -p $HOME/.matplotlib
+   echo "backend : svg" > $HOME/.matplotlib/matplotlibrc
```

Comment:
We seem to consult similar sources :wink:

I have a feeling the suggestion below should work cross platform:
```suggestion
    PYTHON_LIBRARY=`python3 -c "from distutils.sysconfig import get_config_var as gcv; import os.path; print(os.path.join(gcv('LIBDIR'), gcv('INSTSONAME')))"`
    PYTHON_INCLUDE_DIR=`python3 -c "import sysconfig; print(sysconfig.get_path('include'))"`
    echo "--> Detected PYTHON_LIBRARY=$PYTHON_LIBRARY"
    echo "--> Detected PYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR"
    CONFIGURE_PYTHON="-DPYTHON_LIBRARY=$PYTHON_LIBRARY -DPYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR"
    mkdir -p $HOME/.matplotlib
    echo "backend : svg" > $HOME/.matplotlib/matplotlibrc
```